### PR TITLE
ics2115.cpp Updates

### PR DESCRIPTION
**Pull request by cam900**
ics2115.cpp : Implement variable sample rate related actived voice/input clock(Datasheet says standard clock is 33.8688MHz), Implement/Fix Interpolate of u-Law/8bit sample, Add device_rom_interface, Add IO Handler when 16 bit accessing, Add notes, Add imperfect_features() because Envelope/IRQ/Interpolation isn't perfectly emulated

igs_fear.cpp : Add notes
igs011.cpp : Move 16 bit ICS2115 Handler into ics2115.cpp
pgm.cpp : Move CPU_VBLANK_INT_DRIVER into screen VBLANK callback

### CODE REVIEW

File: src/devices/sound/ics2115.cpp [hunk 1]
```diff
@@ -816,6 +879,52 @@ WRITE8_MEMBER(ics2115_device::write)
...
+		case 1:
+			ret = read(space,offset,mem_mask & 0xff);
+			break;
+		case 2:
+		case 3: // TODO : Unknown/Unused at 16 bit accessing?
+			ret = reg_read();
+			break;
+		default:
+#ifdef ICS2115_DEBUG
+			printf("ICS2115: Unhandled memory read at %x\n", offset);
```
Comment (cuavas) [inline src/devices/sound/ics2115.cpp]: Please use `logerror` or things from `logmacro.h` rather than `printf` for diagnostics.

File: src/mame/video/pgm.cpp [hunk 2]
```diff
@@ -669,5 +669,6 @@ WRITE_LINE_MEMBER(pgm_state::screen_vblank_pgm)
 	{
 		/* first 0xa00 of main ram = sprites, seems to be buffered, DMA? */
 		memcpy(m_spritebufferram.get(), m_mainram, 0xa00);
+		m_maincpu->set_input_line(6, HOLD_LINE);
```
Comment (cuavas) [inline src/mame/video/pgm.cpp]: Why was this added?

Comment (cam900) [reply src/mame/video/pgm.cpp]: Some driver has converted VBlank interrupt in screen_vblank callbacks; so I converted VBlank interrupt in here to.

Comment (cuavas) [reply src/mame/video/pgm.cpp]: Oh sorry, I missed where it was removed from mconfig.

**Comment by cuavas** [comment]
Did you actually test this at all?  It fails to build with an unimplemented virtual method.  You really need to test your changes.  Can you please explain why you are still opening pull requests that you haven't tested compile correctly, let alone actually work?
